### ISSUE TITLE
Investigate client crash when using xml printer on managed application

### ISSUE THREAD

**Issue by fmoessbauer** [siemens/drace#10]
```
<Application C:\Program Files\dotnet\dotnet.exe (23768).  Race-Detection Tool 'DRace' internal crash at PC 0x00000262246e2576.  Please report this at https://github.com/siemens/drace/issues.  Program aborted.
0xc0000005 0x00000000 0x00000262246e2576 0x00000262246e2576 0x0000000000000000 0xffffffffffffffff
Base: 0x0000000071000000
Registers: eax=0x01e47a794d520680 ebx=0x00007ff5f3f2e480 ecx=0x00000000000001e7 edx=0x00000262241658c0
        esi=0xffffffffffffffff edi=0x00000000000000c8 esp=0x00007ff5f3f2e3b0 ebp=0x0000000000000001
        r8 =0x00000000000000c8 r9 =0xffffffffffffffff r10=0x00007ff613ea26e4 r11=0x00007ff5f3f6ed5c
        r12=0x00007ff5f3f2eb30 r13=0x00007ff5f3f2ea80 r14=0x0000000000000001 r15=0x0000000000000004
        eflags=0x00007ff500
version 7.1.17990, custom build
```

```
0:007> ~* kb
.  7  Id: 5cd8.888 Suspend: -1 Teb: 000000cb`b4992000 Unfrozen
 # RetAddr           : Args to Child                                                           : Call Site
00 00007ffe`ad5b9252 : 00000000`00000048 00000000`00000001 00000000`00000000 00000000`00000000 : 0x00007ff5`f3ec2e02
01 00007ffe`515f77d8 : 00000262`241aa040 00000000`00000000 00000000`00000000 00000000`00000220 : KERNELBASE!WaitForSingleObjectEx+0xa2

ln 0x00000262246e2576
(00000262`246e2518)   ucrtbase_262246e0000!common_vsnprintf_s<char>+0x5e   |  (00000262`246e2610)   ucrtbase_262246e0000!o__errno
```

**Comment by fmoessbauer** [comment]
Debugging gives some evidence, that the tinyxml printer somehow corrupts the stack. Currently we are unsure if this is a client-transparency issue when printing (`vsnprintf_s`) or a bug in tinyxml.

**Workaround:** Do not use XML printer on managed applications

**Comment by fmoessbauer** [comment]
A in-depth investigation gives some evidence that the actual heap corruption happens in DynamoRIO. Probably the issue just shows up here, as the exporter uses lots of memory allocations.

Xref: see also this entry on the mailing list: https://groups.google.com/forum/#!topic/dynamorio-users/oxRt-CLwMYk

**Comment by fmoessbauer** [comment]
With the fix in 83bf7a905bed32466c2e7729c652cbccd818e7c3 this issue does not seem to occur anymore.
